### PR TITLE
Use Edge Functions instead of Serverless Functions

### PR DESCRIPTION
**Pull request by gregives**

### CODE REVIEW

File: app/api/domains/route.ts [hunk 1]
```diff
@@ -64,25 +57,31 @@ setInterval(() => {
...
+    body: JSON.stringify({
+      model: "gpt-3.5-turbo",
+      messages: [
+        {
+          role: "user",
+          content: `List some suitable domain names for my project in CSV format. Description of my project: "${description}"`,
+        },
+      ],
+    }),
+  }).then((response) => response.json());
```
Comment (gregives) [inline app/api/domains/route.ts:80]: Switched from `openai` package to `fetch` so we can be sure that we're not importing anything that's not supported by Edge Runtime.

File: app/api/domains/route.ts [hunk 2]
```diff
@@ -104,3 +103,5 @@ export async function POST(request: Request) {
 
   return NextResponse.json(availableDomains);
 }
+
+export const runtime = "experimental-edge";
```
Comment (gregives) [inline app/api/domains/route.ts:107]: Not sure if I need to export `config` here instead, or is `config` the old syntax?
```suggestion
export const config = {
  runtime: 'edge',
};
```

File: app/api/domains/route.ts [hunk 3]
```diff
@@ -64,25 +57,31 @@ setInterval(() => {
   currentDomainNameAvailabilityCheck = checkDomainNamesAvailability();
 }, 1000);
 
-export async function POST(request: Request) {
+export async function POST(request: NextRequest) {
```
Comment (gregives) [inline app/api/domains/route.ts:60]: Are POST requests supported at the Edge?

Comment (leerob) [reply app/api/domains/route.ts:60]: Yes, you can use any of the HTTP verbs: https://beta.nextjs.org/docs/routing/route-handlers

File: app/api/domains/route.ts (hunk 2 above)
Comment (leerob) [reply app/api/domains/route.ts:108]: `config` is for `pages/*` – you have the correct syntax.

**Pull request merged**